src/parser/declaration.rs

```diff
@@ -0,0 +1,273 @@
+        match parser.tokens.next() {
+            Some(token_at) => match &token_at.token {
+                Token::Identifier(key) => match key.as_str() {
+                    "background-color" => {
+                        parser.consume_colon_separator()?;
+                        Ok(Declaration::BackgroundColor(parser.parse()?))
+                    }
+                    "border-color" => {
+                        parser.consume_colon_separator()?;
+                        Ok(Declaration::BorderColor(parser.parse()?))
+                    }
+                    "opacity" => {
+                        parser.consume_colon_separator()?;
+                        let opacity = parse_num(parser, 0.0, 1.0)?;
+                        Ok(Declaration::Opacity(opacity))
+                    }
+                    "font-family" => {
+                        parser.consume_colon_separator()?;
+                        Ok(Declaration::FontFamily(parser.parse()?))
+                    }
+                    "font-size" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::FontSize(length))
+                    }
+                    "height" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::Height(length))
+                    }
+                    "width" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::Width(length))
+                    }
+                    "margin" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::Margin(length))
+                    }
+                    "padding" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::Padding(length))
+                    }
+                    "border-width" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::BorderWidth(length))
+                    }
+                    "border-radius" => {
+                        parser.consume_colon_separator()?;
+                        let length: Length = parser.parse()?;
+                        Ok(Declaration::BorderRadius(length))
+                    }
+                    "text-align" => {
+                        parser.consume_colon_separator()?;
+                        let text_al = parser.parse()?;
+                        Ok(Declaration::TextAlign(text_al))
+                    }
```

Comment:
for some of these you use `let` and make a variable that you imediatly use but for others you use `parser.parse()?` directly. You should pick one and be consistant